network-config-analyzer/nca.py

```diff
@@ -69,20 +69,23 @@ def _g_tick():
         time.sleep(next(sleep_for))
 
 
-def _execute_single_config_query(query_name, np1_list, peer_container, output_config):
+def _execute_single_config_query(query_name, np1_list, peer_container, output_config, expected_output=None):
     """
     Runs a query on single set of policies
     :param str query_name: the name of the arg.query
     :param str np1_list: set of policies
     :param PeerContainer peer_container: set of peers
     :param OutputConfiguration output_config: dict object
+    :param str expected_output: a file path to the expected output
     :rtype: int
     """
     network_config1 = NetworkConfig(np1_list, peer_container, [np1_list])
-    return NetworkConfigQueryRunner(query_name, [network_config1], output_config).run_query() > 0
+    res, comparing_err = NetworkConfigQueryRunner(query_name, [network_config1], expected_output,
+                                                  output_config).run_query()
+    return res + comparing_err
```

Comment:
Do you think we need the older version of  `(res > 0) + comparing_err` here because we call `run_query `with `cmd_line_flag=False` on a single config query, whereas for pair configs query we call it with `cmd_line_flag=True`?
I noticed that ` compute_query_output returns` the `bool_result` if `cmd_line_flag `is `True `for semantic diff query. So if the return value for cli queries is the boolean results, I assume we do need the `res>0` version for the single config query?

Reply:
I guess when I wrote the res > 0 , I really  had a case of cmdline query that returned a numerical res = 2 but its expected was 1 , so I concluded to make it a boolean expr (res > 0).
but now we don't have such case, so I am not sure if we still need the res > 0.
( maybe then the test returned 2 instead of 1 because of the changes I was making to code and  I had a wrong conclusion )

Reply:
Which test returned a numerical res = 2  but its expected res was 1? 
I tried to check a different cli query based on a scheme test from `tests\calico_testcases\example_policies\calico-demo\sanity3-scheme.yaml`: After converting this test to a cli query, I see that on this branch the return value is 2, whereas on master branch the return value is True.
I think we should be consistent with return values as before this new feature of "expected output".

Reply:
I can't remember which test it was, but I agree that its better to keep res > 0 
(even though I hesitated yesterday), I believe then I was much concentrating and had much more thinking on it.
If it gave True on master , then the "old" CLExecute file dealt with queries results as bool, so we should keep it that way